### PR TITLE
arm review: definition and assignment command categories should be sxs

### LINKED ISSUE #1020: [help]: 'role' commands miss short summaries

### PR DESCRIPTION
**Pull request by yugangw-msft**
fix partial #1033. I moved definition related commands to under `role definition` or `resource policy definition`. Also fix some argument naming and completer related issues
fix #1020

### CODE REVIEW

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py [hunk 1]
```diff
@@ -342,17 +342,17 @@ def list_policy_assignment(show_all=False, include_inherited=False,
...
+def _build_policy_scope(subscription_id, resource_group_name, scope):
     subscription_scope = '/subscriptions/' + subscription_id
-    if resource_id:
+    if scope:
         if resource_group_name:
-            err = 'Resource group "{}" is redundant because resource id is supplied'
+            err = "Resource group '{}' is redundant because 'scope' is supplied"
             raise CLIError(err.format(resource_group_name))
-        scope = resource_id
+        scope = scope
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: `scope = scope`?
This isn't needed right?

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/generated.py [hunk 2]
```diff
@@ -98,8 +98,8 @@
 cli_command('resource policy assignment list', list_policy_assignment)
 cli_command('resource policy assignment show', show_policy_assignment)
 factory = lambda _: _resource_policy_client_factory().policy_definitions
-cli_command('resource policy create', create_policy_definition)
-cli_command('resource policy delete', PolicyDefinitionsOperations.delete, factory)
-cli_command('resource policy list', PolicyDefinitionsOperations.list, factory)
-cli_command('resource policy show', PolicyDefinitionsOperations.get, factory)
-cli_command('resource policy update', update_policy_definition)
+cli_command('resource policy definition create', create_policy_definition)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/generated.py]: Is this going to create a group that only has one subgroup and no commands? If so, we should use the hyphenated version `resource policy-definition ...` instead. If `resource policy` has commands of its own then this is fine.

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/generated.py [hunk 3]
```diff
@@ -20,10 +20,10 @@
                      _auth_client_factory, _graph_client_factory)
 
 factory = lambda _: _auth_client_factory().role_definitions
-cli_command('role list', list_role_definitions)
-cli_command('role delete', delete_role_definition)
-cli_command('role create', create_role_definition)
-cli_generic_update_command('role update',
+cli_command('role definition list', list_role_definitions)
```
Comment (tjprescott) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/generated.py]: Same comment here. If there are no commands under role, only the definition subgroup, then we should use a hyphenated version `role-definition` instead.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/generated.py (hunk 2 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/generated.py]: The PR just moved all policy definition commands from `az resource policy` to `az resource policy definitions`. So there are no commands under `az resource policy` now, rather just 2 groups, `defintion` and `assignment`. ARM team prefers this layout, and i agree.

File: src/command_modules/azure-cli-role/azure/cli/command_modules/role/generated.py (hunk 3 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-role/azure/cli/command_modules/role/generated.py]: Same with `resource policy`, role has 2 sub-groups `definition` and `assignment`

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py (hunk 1 above)
Comment (yugangw-msft) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py]: fixed

**Pull request merged**